src/audio/module_adapter/module/volume/volume.c

```diff
@@ -252,6 +252,9 @@ static inline int32_t volume_windows_fade_ramp(struct vol_data *cd, int32_t ramp
 	int32_t pow_value; /* Q2.30 */
 	int32_t volume_delta = cd->tvolume[channel] - cd->rvolume[channel]; /* Q16.16 */
 
+	if (!cd->initial_ramp)
```

Comment:
Btw, does it makse sense to always initialse this to a sane non 0 value during new() and reset() so that value is always good and a ramp is always perfomed i.e. no audible artifacts ? 
@singalsu pls comment here.

Reply:
at least for IPC4, this value always comes in module init_instance IPC (curve_duration), so initializing it with non-zero value does not change anything, it is overwritten before use anyway.

Reply:
ok, so this means we need are not validating this value from the IPC ? i.e. host is sending an illegal value ?

Reply:
It is not illegal, it just means "change volume immediately (time=0)"

Reply:
ok, that's fine.

Reply:
another way is:
cd->vol_min and cd->vol_max, if same also means set to target volume immediately.
return cd->vol_min may also works.